SleepEngine/Engine/EntityManager.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Tihran Katolikian
+
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class EntityManager
+{
+    FORBID_COPY_AND_MOVE(EntityManager)
+public:
+    using EntitiesContainer = std::vector<std::unique_ptr<Object>>;
+
+    EntityManager() = default;
+
+    void addObject(std::unique_ptr <Object>&& object, bool shouldUpdate = true);
+    void removeObjectLater(Object const* object);
+    void clear();
+
+    void update(float dt);
+
+private:
+    void postUpdate();
+
+    void addObjects();
+    void removeObjects();
+
+    EntitiesContainer m_objects;
```

Comment:
Is the alias needed? Used in 1 place, impl detal. Makes it harder to understand code.

Reply:
Fixed